utilities/titandb/db_impl_files.cc

```diff
@@ -3,17 +3,20 @@
 namespace rocksdb {
 namespace titandb {
 
-void TitanDBImpl::PurgeObsoleteFiles() {
+std::pair<bool, int64_t> TitanDBImpl::PurgeObsoleteFiles(bool aborted) {
   Status s;
   ObsoleteFiles obsolete_files;
-  vset_->GetObsoleteFiles(&obsolete_files);
+  {
+    MutexLock l(&mutex_);
+    auto oldest_sequence = GetOldestSnapshotSequence();
```

Comment:
It seems to me `GetOldestSnapshotSequence()` can be called without holding `mutex_`.

Reply:
```
  // The lock sequence must be Titan.mutex_.Lock() -> Base DB mutex_.Lock()
  // while the unlock sequence must be Base DB mutex.Unlock() ->
  // Titan.mutex_.Unlock() Only if we all obey these sequence, we can prevent
  // potential dead lock.
  port::Mutex mutex_;
```
We will hold base_db_lock in `GetOldestSnapshotSequence`, so get Titan's lock here

Reply:
I think the comment only applies when we want to lock both of TitanDB mutex and base DB's mutex simultaneously? In this case we release one lock before locking the other, and I think it is fine.

Reply:
Agree, removing that mutex won't lead to a possible deadlock.